File: frontend/src/features/patient/hooks/GetCurrentPatient/GetCurrentPatientDto.ts
import type { Address, EmergencyContact } from "@/features/auth";
import type { LookupDto } from "../LookupDto";
import type { MobilityStatus } from "@/features/patient/types/enums.types";

export interface GetCurrentPatientDto {
  id: string;
  userId: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  dateOfBirth: string;
  gender: string;
  address: Address;
  emergencyContact: EmergencyContact;
  mobilityStatus?: MobilityStatus;
  allergies: LookupDto[];
  conditions: LookupDto[];
  medications: LookupDto[];
  bio?: string;
  profilePictureUrl?: string;
}

File: backend/Modules/Patients/Modules.Patients.Features/Auth/GetCurrentPatient/GetCurrentPatientDto.cs
using Modules.Common.Features.ValueObjects;
using Modules.Patients.Domain.Enums;
using Modules.Patients.Domain.ValueObjects;
using Modules.Patients.Features.GetAllergies;
using Modules.Patients.Features.GetConditions;
using Modules.Patients.Features.GetMedications;

namespace Modules.Patients.Features.Auth.GetCurrentPatient;

public sealed record GetCurrentPatientDto(
    Guid Id,
    Guid UserId,
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    string DateOfBirth,
    string Gender,
    Address Address,
    EmergencyContact EmergencyContact,
    MobilityStatus? MobilityStatus,
    List<AllergyDto> Allergies,
    List<ConditionDto> Conditions,
    List<MedicationDto> Medications,
    string? Bio,
    string? ProfilePictureUrl);

